ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/your_flist.py

```diff
@@ -99,9 +103,30 @@ def container_farm(self):
         else:
             query["hru"] = storage_units
         farms = j.sal.reservation_chatflow.farm_names_get(1, self, currency=self.currency, **query)
-        self.node = j.sal.reservation_chatflow.nodes_get(
-            1, farm_names=farms, currency=self.currency, **query
-        )[0]
+        self.node = j.sal.reservation_chatflow.nodes_get(1, farm_names=farms, currency=self.currency, **query)[0]
+
+    @j.baseclasses.chatflow_step(title="Attach Volume")
+    def container_volume(self):
+        volume_attach = self.drop_down_choice(
+            "Would you like to attach an extra volume to the container", ["YES", "NO"], required=True, default="NO"
+        )
+        self.container_volume_attach = volume_attach == "YES" or False
+
+    @j.baseclasses.chatflow_step(title="Volume details")
+    def container_volume_details(self):
+        if self.container_volume_attach:
+            form = self.new_form()
+            vol_disk_type = form.drop_down_choice(
+                "Please choose the type of disk for the volume", ["SSD", "HDD"], required=True, default="SSD"
+            )
+            vol_disk_size = form.int_ask(
+                "If you need an extra volume to be attached to the container, please specify its size",
+                required=True,
+                default=10,
+            )
+            form.ask()
+            self.user_form_data["Volume Disk type"] = vol_disk_type.value
+            self.user_form_data["Volume Size"] = vol_disk_size.value
```

Comment:
please allow them to choose the mountpoint aswell (you can default to /data)

Reply:
okay

Reply:
Added here : https://github.com/threefoldtech/jumpscaleX_threebot/pull/746/commits/bcb5fdc579cb1b5b38b51a942a628ee936230afe